### PR TITLE
nspace.d: use foreachDsymbol()

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Comment by jacob-carlborg** [comment]
Why the mix of lambda and delegate syntax?

File: src/dmd/nspace.d [hunk 1]
```diff
@@ -127,50 +120,21 @@ extern (C++) final class Nspace : ScopeDsymbol
...
             dsymbolSemantic(this, null); // try to resolve it
-        if (members)
-        {
-            foreach (s; *members)
-            {
-                //printf("\t%s\n", s.toChars());
-                s.setFieldOffset(ad, poffset, isunion);
-            }
-        }
+        members.foreachDsymbol( s => s.setFieldOffset(ad, poffset, isunion) );
```
[the remark this answers is not in the document]
Comment (WalterBright) [reply src/dmd/nspace.d]: I like the extra spaces in this context.

**Comment by WalterBright** [comment]
> Why the mix of lambda and delegate syntax?

As mentioned several times now, the latter is to avoid emitting more than one copy of the same function.

**Review by thewilsonator — APPROVED** [review]

**Comment by jacob-carlborg** [comment]
> As mentioned several times now, the latter is to avoid emitting more than one copy of the same function

When does that occur?

**Pull request merged**